openlcs/libs/corgi.py

```diff
@@ -238,3 +276,94 @@ def get_paginated_data(self, query_params=None, api_path="components"):
                 # general exceptions, timeout/connection/maxredirect etc.
                 logger.error("Request failed: %s", e)
                 break
+
+    def get_srpm_component(self, component):
+        """
+        Returns the source rpm component for `component`
+        """
+        if component["arch"] == "src":
+            return (True, component)
+        sources = component.get("sources")
+        if not sources:
+            # This is not likely to happen since corgi promises
+            # to always have corresponding srpm.
+            return (False, component.get("purl"))
+        link = find_srpm_source(sources)
+        if link:
+            logger.info("Querying component: %s", link)
+            component = self.get(link)
+            return (True, component) if component else (False, link)
+        else:
+            return (False, component.get("purl"))
+
+    def fetch_component(self, link):
+        component = self.get(link)
+        if component:
+            if component.get("type") == "RPM":
+                return self.get_srpm_component(component)
+            # FIXME: examine what to return for non-rpm components.
+            return (True, component)
+        else:
+            return (False, link)
+
+    def get_container_source_components(self, component):
+        """
+        Extract source components from a corgi container component
+
+        If the name of the component ends with "-source", it will be handled
+        separately. Otherwise the function looks for binary components
+        provided by the component and retrieves their corresponding
+        source components. The source components are returned as a list.
+        There are chances that after several retries the component retrieval
+        still fails, failed ones are returned as a list of strings.
+
+        Args:
+        component (dict): The OCI component info
+
+        Returns:
+        Tuple: (components found / missings)
+        """
+        name = component.get("name")
+        if name.endswith("-source"):
+            logger.debug("This is a source container build")
+            sources = component.get("sources")
+            if not sources:
```

Comment:
I found almost all the source containers in the corgi, the `sources` field is empty